File: examples/breakout_pmw3901/motion/motion.cpp
#include <stdio.h>
#include "pico/stdlib.h"

// Uncomment the below line to switch from the PMW3901 to the PAA5100
//#define USE_PAA5100

#ifndef USE_PAA5100
  #include "breakout_pmw3901.hpp"
#else
  #include "breakout_paa5100.hpp"
#endif

using namespace pimoroni;

#ifndef USE_PAA5100
  typedef BreakoutPMW3901 FlowSensor;
#else
  typedef BreakoutPAA5100 FlowSensor;
#endif

FlowSensor flo(BG_SPI_FRONT);

int main() {
  stdio_init_all();

  flo.init();
  flo.set_rotation(FlowSensor::DEGREES_0);
  int16_t tx = 0, ty = 0;
  int16_t x = 0, y = 0;
  while(true) {
    if(flo.get_motion(x, y)) {
      tx += x;
      ty += y;
      printf("Relative: x %6d, y %6d | Absolute: tx %6d, ty %6d\n", x, y, tx, ty);
    }
    sleep_ms(10);
  };
  return 0;
}

File: micropython/examples/breakout_pmw3901/motion.py
import time

# Pick *one* sensor type by uncommenting the relevant line below:

# PMW3901
from breakout_pmw3901 import BreakoutPMW3901 as FlowSensor

# PAA5100
# from breakout_paa5100 import BreakoutPAA5100 as FlowSensor

flo = FlowSensor()
flo.set_rotation(FlowSensor.DEGREES_0)

tx = 0
ty = 0
x = 0
y = 0

while True:
    delta = flo.get_motion()
    if delta is not None:
        x = delta[0]
        y = delta[1]
        tx += x
        ty += y
        print("Relative: x {}, y {} | Absolute: tx {}, ty {}".format(x, y, tx, ty))
    time.sleep(0.1)
